crowbar_framework/app/models/service_object.rb

```diff
@@ -1852,4 +1852,25 @@ def node_changed_roles?(node, old_role, new_role)
     roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
     roles_in_old != roles_in_new
   end
+
+  # return true if no other attribute than the ignored has changed. And that the node has no other
+  # role than the roles in only_for_roles
+  def relevant_attributes_changed_if_roles?(node, old_role, new_role, ignore_attributes, only_for_roles)
+    # sort roles of only_for_roles
+    only_for_roles = only_for_roles.sort
+
+    # return true if node has no other roles that ones in the list (only_for_roles)
+    roles_in_old = old_role.elements.keys.select { |r| old_role.elements[r].include?(node) }.sort
```

Comment:
thinking out loud here, should we care if the node did not had the role before?

I mean, if it changed roles, it will trigger a run, so we may assume that it had the role before otherwise we would not be reaching this code, so maybe we can drop this?

In any case, it does not hurt

Reply:
I was too late on this as I see that Vincent already mentions this :)

Reply:
Yes, I didn't be sure about that. Thanks!
Sort was to optimize because other ways I had to sort twice the same list.